### ISSUE TITLE
[Question] Bot side default configuration without renovate config in most repos

### ISSUE THREAD

**Issue by cobexer** [renovatebot/config-help#429]
### Which Renovate are you using? CLI, App, or Pro
Pro

### Which platform are you using? GitHub, GitLab, Bitbucket Azure DevOps
GitLab

### What is your question?
I want to configure renovate on the bot side without renovate config in most repositories. Onboarding should just be giving the GitLab bot user access to a new repository.
I found the `force` property in the configuration - but I want to be able to allow repositories to override the company wide default configuration if necessary. Also it appears as if the `hostRules` set using `force` were not properly applied even tough `printConfig` showed them.

TL;DR: I want to configure everything (save exceptions) on the bot side in config.js - no onboarding PR's, no renovate config in repos

**Comment by rarkins** [comment]
There is no need to use `force` and I recommend that you don't. Force is intended for extreme circumstances and not intended for nested config like hostRules either.

If you put config in `config.js` (no `force` necessary) then it will be shared amongst every repo and they can override it with their own repo config, but the downside is that your repo users won't be able to "see" the config they need to override.

A better approach would be to configure a "preset" config that is shared amongst repositories, and then set that as `onboardingConfig`. This way you can have a central config that is inherited by each repo while also being able to be overrided, meanwhile users can view what's inside it too. It works with npm modules and should be possible to work with local GitLab repos too, but I haven't tested that to be sure. It could certainly be made to work with a little code fix if necessary.

**Comment by cobexer** [comment]
> There is no need to use `force` and I recommend that you don't. Force is intended for extreme circumstances and not intended for nested config like hostRules either.

Yes I definitely don't want to use `force`

> If you put config in `config.js` (no `force` necessary) then it will be shared amongst every repo and they can override it with their own repo config, but the downside is that your repo users won't be able to "see" the config they need to override.

Users would not be allowed to override the config anyway

> A better approach would be to configure a "preset" config that is shared amongst repositories, and then set that as `onboardingConfig`. This way you can have a central config that is inherited by each repo while also being able to be overrided, meanwhile users can view what's inside it too. It works with npm modules and should be possible to work with local GitLab repos too, but I haven't tested that to be sure. It could certainly be made to work with a little code fix if necessary.

Avoiding an `onboardingConfig` with a pointer to a GitLab repo/NPM package with the presets would be very nice!
Currently we use a config like this:
```
module.exports = {
  "onboarding": false,
  "printConfig": false,
  "requireConfig": false,
  "npmrc": "registry = https://artifactory.example.com/artifactory/api/npm/npm\n_auth = XXX==\nalways-auth = true",
  "hostRules": [{
    "domainName": "artifactory.example.com",
    "username": "user",
    "password": "password"
  }],
  "gradle-wrapper": { "enabled": false },
  "prHourlyLimit": 0,
  "prConcurrentLimit": 0
}
```

But we get errors indicating that `hostRules` don't work:
```
WARN: Dependency lookup unauthorized. Please add authentication with a hostRule (repository=cd/infrastructure-test)
       "pkgUrl": "https://artifactory.example.com/artifactory/dependencies/junit/junit/maven-metadata.xml"
```

Considering that everything internally requires authentication we definitely need to manage `hostRules` centrally in the bot config.

Any ideas why this isn't working?

**Comment by rarkins** [comment]
Which version of Pro are you trying? Can you try the very latest 0.18.0 pre-release image? There was a hostRules problem fixed in the recent pre-release that might help you.

**Comment by cobexer** [comment]
I did of course use the latest stable tag (0.17.0) using 0.18.0 pre 4 fixes the hostRules 😃

Thanks!
